File: app/src/main/java/com/craiovadata/android/sunshine/ui/main/ViewModelFactory.java
package com.craiovadata.android.sunshine.ui.main;

import com.craiovadata.android.sunshine.data.RepositoryWeather;

import org.jetbrains.annotations.NotNull;

import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

public class ViewModelFactory extends ViewModelProvider.NewInstanceFactory {

    private final RepositoryWeather mRepository;

    public ViewModelFactory(RepositoryWeather repository) {
        this.mRepository = repository;
    }

    @NotNull
    @Override
    public <T extends ViewModel> T create(@NotNull Class<T> modelClass) {
        //noinspection unchecked
        return (T) new MainViewModel(mRepository);
    }

}

File: app/src/main/java/com/craiovadata/android/sunshine/ui/main/ViewModelFactory.kt
package com.craiovadata.android.sunshine.ui.main

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider.NewInstanceFactory
import com.craiovadata.android.sunshine.data.database.Repository

class
ViewModelFactory(private val mRepository: Repository) :
    NewInstanceFactory() {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return MyViewModel(mRepository) as T
    }

}